### PR TITLE
Completed week 4 assignment: moved frontend files to frontend-public …

### PR DESCRIPTION
**Pull request by AidaBur**
…directory, updated static file serving path in app.js, refactored routes and controllers for managing people data, added functionalities for adding, getting, updating, and deleting people.

### CODE REVIEW

File: 02-express-tutorial/app.js [hunk 1]
```diff
@@ -1,15 +1,40 @@
 const express = require("express");
 const path = require("path");
-const { products } = require("./data");
-console.log("Express Tutorial");
+const { products, people } = require("./data");  
+const peopleRouter = require("./routes/people");
```
Comment (chase-cove) [inline 02-express-tutorial/app.js:4]: The `peopleRouter` is imported here but not used

File: 02-express-tutorial/controllers/people.js
Comment (chase-cove) [inline 02-express-tutorial/controllers/people.js:1]: this looks great!

File: 02-express-tutorial/app.js [hunk 2]
```diff
@@ -50,14 +76,35 @@ app.get("/api/v1/query", (req, res) => {
...
+  const newPerson = {
+    id: people.length + 1,  
+    name: req.body.name,   
+  };
+
+  people.push(newPerson);
+
+  
+  res.status(201).json({ success: true, name: req.body.name });
 });
```
Comment (chase-cove) [inline 02-express-tutorial/app.js]: the refactoring is not quite complete here. this is where we're just wanting to use the `peopleRouter` to replace the initial people routes here

File: 02-express-tutorial/routes/people.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// routes/people.js
+const express = require("express");
+const router = express.Router();
+
+const { addPerson, getPeople, getPersonById, updatePerson, deletePerson } = require("../controllers/people.js");
+
+router.get("/", getPeople);
+
+router.post("/", addPerson);
+
```
Comment (chase-cove) [inline 02-express-tutorial/routes/people.js:10]: looks good!

File: 02-express-tutorial/frontend-public/index.html.save
Comment (chase-cove) [inline 02-express-tutorial/frontend-public/index.html.save:1]: Did you want to create this file?

**Review by chase-cove — CHANGES_REQUESTED** [review]

**Comment by chase-cove** [comment]
Hi Aida,

This is looking really good! The people router and controller files look great. It just looks like the final bit of refactoring hasn't happened yet. `peopleRouter` is imported into `app.js` but isn't being used and just needs to replace those two initial people routes

File: 02-express-tutorial/app.js (hunk 1 above)
Comment (AidaBur) [reply 02-express-tutorial/app.js:4]: I utilized the imported peopleRouter by adding it to app.js with app.use("/api/v1/people", peopleRouter) to ensure it's used for handling the relevant routes.

File: 02-express-tutorial/app.js (hunk 2 above)
Comment (AidaBur) [reply 02-express-tutorial/app.js]: I replaced the initial /api/v1/people routes (such as app.post, app.get, etc.) with the peopleRouter to handle all these routes and ensure better modularity and code organization.

File: 02-express-tutorial/app.js [hunk 4]
```diff
@@ -30,10 +30,13 @@ app.get("/", (req, res) => {
...
 
 
-app.get("/api/v1/products", (req, res) => {
+app.use("/api/v1/products", (req, res) => {
```
Comment (chase-cove) [inline 02-express-tutorial/app.js]: Hey, this is so close! But we just need to do it for the main people route and use the peopleRouter imported above

```
app.use("/api/v1/people", peopleRouter);
```

**Comment by chase-cove** [comment]
Hi Aida,

Just a small change needed for  the `app.use` for the people routes, but other than that it looks great!

**Review by chase-cove — APPROVED** [review]

**Pull request merged**